lib/modules/manager/pixi/extract.ts

```diff
@@ -65,8 +76,101 @@ export async function extractPackageFile(
     lockFiles.push(lockfileName);
   }
 
+  const project = val.project;
+
+  const channelPriority = project['channel-priority'];
+
+  let registryStrategy: RegistryStrategy | undefined;
+
+  if (channelPriority === 'disabled') {
+    registryStrategy = 'merge';
+  }
+
+  // resolve channels and build registry urls for each channel with order
+  const conda: PixiPackageDependency[] = [];
+
+  for (const item of val.conda) {
+    conda.push(
+      addRegistryUrls({
+        ...item,
+        channels: project.channels,
+        registryStrategy,
+      }),
+    );
+  }
+
+  for (const item of val.feature.conda) {
+    conda.push(
+      addRegistryUrls({
+        ...item,
+        registryStrategy,
+        channels: [...coerceArray(item.channels), ...project.channels],
+      }),
+    );
+  }
+
   return {
     lockFiles,
-    deps: [],
+    deps: [conda, val.pypi, val.feature.pypi].flat(),
+  };
+}
+
+function addRegistryUrls(item: PixiPackageDependency): PixiPackageDependency {
+  const channels = orderChannels(item.channels);
+
+  if (item.channel) {
+    return {
+      ...item,
+      channels,
+      registryUrls: [channelToRegistryUrl(item.channel)],
+    };
+  }
+
+  if (channels.length === 0) {
+    return {
+      ...item,
+      channels,
+      skipStage: 'extract',
+      skipReason: 'unknown-registry',
+    };
+  }
+
+  const registryUrls: string[] = [];
+  for (const channel of channels) {
+    registryUrls.push(channelToRegistryUrl(channel));
+  }
+
+  return {
+    ...item,
+    channels,
+    registryUrls,
   };
 }
+
+function channelToRegistryUrl(channel: string): string {
+  if (isHttpUrl(channel)) {
+    return ensureTrailingSlash(channel);
+  }
+
+  return `${defaultCondaRegistryApi}${channel}/`;
```

Comment:
```suggestion
  return ensureTrailingSlash(joinUrlParts(defaultCondaRegistryApi, channel));
```
If somebody tries to inject code into the channel string.